python/tank/platform/interactive_authentication.py

```diff
@@ -295,28 +299,38 @@ def ui_renew_session():
     """
     Prompts the user to enter his password in a dialog to retrieve a new session token.
     """
-    UiAuthenticationHandler(is_session_renewal=True).authenticate()
+    # Force human user authentication, since ression renewal is always in the context of a user login.
+    UiAuthenticationHandler(is_session_renewal=True).authenticate(force_human_user_authentication=True)
 
 
-def ui_authenticate():
+def ui_authenticate(force_human_user_authentication=False):
     """
-    Prompts the user to login via a dialog and caches the session token for future reuse.
+    Authenticates the current process. Authentication can be done through script user authentication
+    or human user authentication. If doing human user authentication and there is no session cached, a
+    dialgo asking for user credentials will appear.
```

Comment:
I wonder if the parameter `force_human_user_authentication` could be named differently - i must admit i am a little confused between user authentication, renewal etc. I thought session renewal basically meant "please type in your password" and then "user authentication" would represent the whole thing with username, password etc.

Maybe it should be called `ignore_script_credentials` or something along those lines instead? What do you think?